hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -178,14 +180,16 @@ public BlockOutputStream(
     if (replicationIndex > 0) {
       blkIDBuilder.setReplicaIndex(replicationIndex);
     }
+    this.directBufferPool = directBufferPool;
     this.containerBlockData = BlockData.newBuilder().setBlockID(
         blkIDBuilder.build()).addMetadata(keyValue);
     // tell DataNode I will send incremental chunk list
-    if (config.getIncrementalChunkList()) {
+    if (config.getIncrementalChunkList() && directBufferPool != null) {
```

Comment:
This seems incorrect. If config.getIncrementalChunkList() is true, it requires a buffer to allocated. If directBufferPool is not null, then the buffer will come from this pool, otherwise, allocate a buffer instead.